doc/source/serve/deployment-graph/deployment_graph_intro.md

```diff
@@ -0,0 +1,219 @@
+
+Here's an example:
+
+```{literalinclude} ../doc_code/deployment_graph_intro/class_nodes.py
+:start-after: __hello_start__
+:end-before: __hello_end__
+:language: python
+:linenos: true
+```
+
+In line 40, the `LanguageClassifier` deployment takes in the `spanish_responder` and `french_responder` `ClassNodes` as constructor arguments. Its `__call__` method uses the request's values to decide whether to respond in Spanish or French. It then forwards the request's name to the `SpanishResponder` or the `FrenchResponder` on lines 17 and 19. The calls are formatted as:
+
+```python
+self.spanish_responder.say_hello.remote(name)
+```
+
+This call has a few parts:
+* `self.spanish_responder` is the `SpanishResponder` node taken in through the constructor.
+* `say_hello` is the `SpanishResponder` method to invoke on this node.
+* `remote` indicates that this is a remote call to another deployment. This is required when invoking a deployment's method through another deployment. It needs to be added to the method name.
+* `name` is the argument for `say_hello`. You can pass any number of arguments or keyword arguments here.
+
+This call returns a reference to the result– not the result itself. This pattern allows the call to execute asynchronously. To get the actual result, call `ray.get` on the result. This function blocks until the asynchronous call executes and then returns the result. In this example, line 23 calls `ray.get(ref)` and returns the resulting string.
+
+You can try this example out using the `serve run` CLI:
+
+```console
+$ serve run hello:language_classifier
+```
+
+You can use this client script to interact with the example:
+
+```{literalinclude} ../doc_code/deployment_graph_intro/class_nodes.py
+:start-after: __hello_client_start__
+:end-before: __hello_client_end__
+:language: python
+```
+
+While the `serve run` is running, open a separate terminal window and run this script:
+
+```console
+$ python hello_client.py
+
+Hola Dora
+```
+
+(deployment-graph-drivers-http-adapters-intro)=
+## Drivers and HTTP Adapters
+
+Ray Serve provides the `DAGDriver`, which routes HTTP requests through your call graph. As mentioned in [the call graph section](deployment-graph-intro-call-graph), the `DAGDriver` has one required argument: the `FunctionNode` or `MethodNode` representing your call graph's final output.
+
+The `DAGDriver` also has one more optional keyword argument: `http_adapter`. `http_adapters` are functions that get run on the HTTP request before it's passed into the graph. Ray Serve provides a handful of these adapters, so you can rely on them to conveniently handle the HTTP parsing while focusing your attention on the graph itself.
+
+For instance, we can use the Ray Serve-provided `json_request` adapter to simplify our [arithmetic call graph](deployment-graph-intro-arithmetic-graph) by eliminating the `unpack_request` function. Here's the revised call graph and driver:
```

Comment:
Nice, it might be good to add a tip box linking to this section somewhere in the call graph section.  Saying something like "if you don't want to do this unpacking manually, jump to [this section]"

Reply:
Good idea, I added a tip.